build

```diff
@@ -81,6 +81,12 @@ echo "$ROSDEP_SOURCE" > "$APT_REPO/2-remote.list"
 
 ROS_HOME="$APT_REPO/ros" ROSDEP_SOURCE_PATH="$APT_REPO:/etc/ros/rosdep/sources.list.d/" rosdep update
 
+
+if [ -n "$PRE_BUILD_HOOK" ]; then
+  echo "Run pre-build hook"
+  eval "$PRE_BUILD_HOOK"
```

Comment:
I wonder if it is better to run a script here.

Reply:
I had the same thought in mind, the user specifies a path to a script that get executed rather than a string. Do you want me to update it?

Reply:
would be great if you could do it.